assets/javascripts/kitten/hoc/modal.js

```diff
@@ -0,0 +1,76 @@
+import React, { Component } from 'react'
+// Via "https://github.com/reactjs/react-modal"
+import ReactModal from 'react-modal'
+import { CloseButton } from 'kitten/components/buttons/close-button'
+
+export const modal = (ModalContentComponent, TriggerComponent) => {
```

Comment:
J'ai l'impression qu'on a un hoc hyper compliqué pour cette modale.

Ça implique de créer deux composants (`ModalContentComponent` et `TriggerComponent`) à chaque fois qu'on veut faire une modale différente. Si on veut ajouter une modale on doit faire tout ça :

```jsx
const ErrorContent = () => (
    <p>Error</p>
)
const ErrorButton = (props) => (
    <Button tag="a" { ...props } />
)
const ErrorModal = modal(ErrorContent, ErrorButton)

const InfoLine = () => (
  <div>Info <AlertModal /></div>
)
```

Peut-être qu'on peut faire plus simple. Par exemple @LanF3usT qu'on utilise pas de `hoc` mais qu'on ait juste un composant qui accepte une prop `trigger`.

```js
const InfoLine = () => (
  <p>
    Info
    <Modal trigger={ <Button tag="a">Click Me</Button> }>
      <p>Error</p>
    </Modal>
  </p>
)
```

Pour ça, au lieu d'ajouter `onClick` sur le composant `trigger` on pourrait le mettre *autour*, par exemple avec `<span onClick="…">{ this.props.trigger }</span>`.